controllers/devworkspace/solver/che_routing.go

```diff
@@ -342,14 +362,43 @@ func (c *CheRoutingSolver) getGatewayConfigsAndFillRoutingObjects(cheCluster *ch
 	return configs, nil
 }
 
-func (c *CheRoutingSolver) getInfraSpecificExposer(cheCluster *chev2.CheCluster, routing *dwo.DevWorkspaceRouting, objs *solvers.RoutingObjects) (func(info *EndpointInfo), error) {
+func getNormalizedUsername(c client.Client, namespace string) (string, error) {
+	secret := &corev1.Secret{}
+	err := c.Get(context.TODO(), client.ObjectKey{Name: "user-profile", Namespace: namespace}, secret)
+	if err != nil {
+		return "", err
+	}
+	username := string(secret.Data["name"])
+	return normalize(username), nil
```

Comment:
The username is being read from the `user-profile` secret created by che-server.

Another way to read the username could be to read the `che.eclipse.org/username` annotation from the user namespace

Reply:
I'm afraid there is no backup plan right now if that secret does not exist, are there other ways to detect the username?

Reply:
What about using the legacy route name if username is unknown?

Reply:
After more investigation, I think the minikube tests are failing because the `user-profile` secret is not created when the user namespace and workspace is created by applying templates (ie, without using the dashboard)

Reply:
@dkwon17 thanks for the investigation, so in this case the flow when devworkspace is created using `oc apply` command is going to also fail if there is no user-profile secret in the namespace, right?

Reply:
@ibuziuk yes, that's right

Reply:
I am currently working on the legacy routes in this case, as suggested by @tolusha